src/main/java/io/split/android/client/storage/splits/SplitEntityToSplitTransformer.java

```diff
@@ -59,23 +59,28 @@ private List<SplitDeferredTaskItem<List<Split>>> getSplitDeserializationTasks(Li
 
         for (List<SplitEntity> partition : partitions) {
             taskList.add(new SplitDeferredTaskItem<>(
-                    () -> convertEntitiesToSplitList(partition)));
+                    () -> convertEntitiesToSplitList(partition, mSplitCipher)));
```

Comment:
Please check that you're not using a single instance of Android cipher in multiple thread